app/domain/authentication/util/fetch_authenticator_secrets.rb

```diff
@@ -0,0 +1,38 @@
+require 'command_class'
+require 'conjur/fetch_required_secrets'
+
+module Authentication
+  module Util
+
+    FetchAuthenticatorSecrets = CommandClass.new(
+      dependencies: {
+        fetch_secrets: ::Conjur::FetchRequiredSecrets.new
+      },
+      inputs:       %i(conjur_account authenticator_name service_id required_variable_names)
+    ) do
+
+      def call
+        secrets = {}
+
+        required_secrets = @fetch_secrets.(resource_ids: required_resource_ids)
+
+        @required_variable_names.each do |variable_name|
+          full_variable_name     = full_variable_name(variable_name)
+          secrets[variable_name] = required_secrets[full_variable_name]
+        end
+
+        secrets
+      end
```

Comment:
This can be cleaned up a bit and made more consistent with the style we use elsewhere (very simple call, named memoized methods for any relevant concept.  This should work (untested though):
```ruby
#... 
      def call
        @required_variable_names.reduce({}) do |secrets, variable_name|
          full_variable_name     = full_variable_name(variable_name)
          secrets[variable_name] = required_secrets[full_variable_name]
          secrets
        end
      end

      private

      def required_secrets
        @required_secrets ||= @fetch_secrets.(resource_ids: required_resource_ids)
      end
#... 
```

Reply:
Since you're using `each_with_object` rather than `reduce` (which is a good call) you can delete line 18: `secrets`.